src/mixins/document-mixin.ts

```diff
@@ -71,19 +83,31 @@ export default class DocumentMixin extends Vue {
   }
 
   /**
-   * Deletes a Minio document from Legal API.
-   * @param documentKey the document key
+   * Deletes a Minio document via Legal API.
+   * @param key the document key
+   * @param deleteKey whether to delete the key from the unsaved documents list
    * @returns a promise to return the axios response or the error response
    */
-  async deleteDocument (documentKey: string): Promise<AxiosResponse> {
-    // safety checks
-    if (!documentKey) {
-      throw new Error('Invalid parameters')
+  async deleteDocument (key: string, deleteKey = true): Promise<AxiosResponse> {
+    if (deleteKey) {
+      // delete key from unsaved documents list
+      this.removeUnsavedDocument(key)
     }
 
-    const url = `documents/${documentKey}`
+    return axios.delete(`documents/${key}`)
+  }
 
-    return axios.delete(url)
+  /**
+   * Deletes documents uploaded in this session but where the draft filing, and hence the file keys,
+   * was not saved to the API. This is to prevent orphaned documents in Minio.
+   */
+  async deleteUnsavedDocuments (): Promise<Array<any>> {
+    const promises = []
+    for (const key of this.getUnsavedDocuments) {
+      promises.push(this.deleteDocument(key, false))
+    }
+    // make all API calls concurrently without rejection
+    return Promise.allSettled(promises)
```

Comment:
I tried quite a few things here to "wait" for the Axios delete calls to complete, but:
- the calls aren't processed until after this method runs (because JS is single-threaded)
- the calls are all cancelled because the app terminates right after this